### DISCUSSION TITLE
Integrating mqtt with the cppagent

### DISCUSSION THREAD

**Discussion by Varun-Poojary** [mtconnect/cppagent#313] [Q&A]
Hi, 

I am trying to experiment with MQTT.

I am able to build and run the cppagent using docker
Any suggestions on how to integrate this agent with mqtt.

Thank you.

**Comment by wsobel** [comment]
There is an existing MQTT sink that can be enabled, it just needs to know where the broker is. 

I am working with docker compose to create some standard configuration examples. Check out the `docker_with_shared_libs_volumes` branch for some of the latest. Check out `demo` and `docker` directories`. Still a more to do.

One should be able to take the demo compose add mosquitto broker and then point the mqtt sink at the mosquitto container. 

Best,
W
